### PR TITLE
BLOG: Add post about dataframe interoperability (PyCon Lithuania 2024 talk)

### PR DESCRIPTION
**Pull request by MarcoGorelli**
## Text styling

- [x] The blog is written with [plain language](https://www.plainlanguage.gov/guidelines/) (where relevant).
- [x] If there are headers, they use the proper header tags in order to do so (with only one level-one header).
- [x] All links describe where they link to (for example, check the [Quansight labs website](https://labs.quansight.org/)).
- [x] Any kind of styling that the author uses (for example, bold for emphasis) is consistent throughout the blog.

## Non-text contents

- [x] Blog post featured image is in PNG or JPEG format, **not** SVG.
- [x] All content is represented as text (for example, images need alt text and videos need captions or descriptive transcripts).
- [x] If there are emojis, there are not more than three in a row.
- [x] Don't use [flashing gifs or videos](https://www.w3.org/TR/UNDERSTANDING-WCAG20/seizure-does-not-violate.html).
- [x] If it were to be read as plain text, the blog still makes sense and no information is missing.

### CODE REVIEW

File: apps/labs/posts/dataframe-interop-pycon-lit-2024.md [hunk 1]
```diff
@@ -0,0 +1,180 @@
+---
+title: 'Dataframe interoperability - what has been achieved, and what comes next?'
+authors: [marco-gorelli]
+published: April 9, 2024
```
Comment (rgommers) [inline apps/labs/posts/dataframe-interop-pycon-lit-2024.md:4]:
```suggestion
published: May 21, 2024
```

File: apps/labs/posts/dataframe-interop-pycon-lit-2024.md [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+  src: /posts/dataframe-interop-pycon-lit-2024/narwhals_small.jpg
+  alt: 'Narwhals logo'
+hero:
+  imageSrc: /posts/dataframe-interop-pycon-lit-2024/narwhals_medium.jpg
+  imageAlt: 'Narwhals logo'
+---
+
+# DataFrame Interoperability
+
+I attended PyConLithuania 2024, and had a blast! I don't speak Lithuanian, and probably
```
Comment (rgommers) [inline apps/labs/posts/dataframe-interop-pycon-lit-2024.md:17]:
```suggestion
I attended PyCon Lithuania 2024, and had a blast! I don't speak Lithuanian, and probably
```

File: apps/labs/posts/dataframe-interop-pycon-lit-2024.md [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+Spanish, you might think that "estoy embarazado" means "I'm embarrassed", but it actually
+means "I'm pregnant". Similarly, after learning a bit of pandas and Polars, you might
+expect
+```python
+import pandas as pd
+import polars as pd
+
+print((3 in pd.Series([1,2,3])) == (3 in pl.Series([1,2,3])))
+```
+to print `'True'` - but no so! pandas checks if `3` is in the index, whereas Polars checks
```
Comment (rgommers) [inline apps/labs/posts/dataframe-interop-pycon-lit-2024.md:34]:
```suggestion
to print `'True'` - but not so! pandas checks if `3` is in the index, whereas Polars checks
```

File: apps/labs/posts/dataframe-interop-pycon-lit-2024.md [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+- writing dataframe-agnostic code is hard!
+- a simple and clear common language can enable collaboration
+
+I'll tell you about the dataframe landscape, the interchange protocol, how to write
+dataframe-agnostic code, and where we go from here.
+
+## pandas are everywhere
+
+There's a great array of diverse dataframes out there. And the way that data science libraries
+have typically to such diversity is to support pandas...
```
Comment (rgommers) [inline apps/labs/posts/dataframe-interop-pycon-lit-2024.md:48]:
```suggestion
typically have to such diversity is to support only pandas...
```

File: apps/labs/posts/dataframe-interop-pycon-lit-2024.md [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+
+This little prelude was just to establish two premises:
+
+- writing dataframe-agnostic code is hard!
+- a simple and clear common language can enable collaboration
+
+I'll tell you about the dataframe landscape, the interchange protocol, how to write
+dataframe-agnostic code, and where we go from here.
+
+## pandas are everywhere
```
Comment (rgommers) [inline apps/labs/posts/dataframe-interop-pycon-lit-2024.md:45]:
```suggestion
## pandas is everywhere
```

File: apps/labs/posts/dataframe-interop-pycon-lit-2024.md [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+1. Use `narwhals.from_native` to wrap any dataframe in a Narwhals class.
+2. Use the subset of the Polars API supported by Narwhals.
+3. Use `narwhals.to_native` to return the object to the user in its original class.
+
+If you follow these steps, then congrats, your code will work seamlessly across pandas,
+Polars, Modin, and cuDF. But most importantly, GPU code can stay on GPU, and lazy code
+can stay lazy!
+
+To see a complete example of Narwhals in action, its website features
+[a little tutorial](https://marcogorelli.github.io/narwhals/basics/complete_example/).
```
Comment (rgommers) [inline apps/labs/posts/dataframe-interop-pycon-lit-2024.md:130]: this link is broken, as are the other ones to this site

File: apps/labs/posts/dataframe-interop-pycon-lit-2024.md [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+of them:
+
+- Users need not do any data conversion manually (in fact, neither does Narwhals, it only
+  translates syntax).
+- Data which starts on GPU can stay on GPU.
+- Data which starts lazy can stay lazy.
+- No heavy dependencies necessary (as we'll soon see, Narwhals is about as light as it gets).
+
+Finally, [Narwhals is also extensible](https://marcogorelli.github.io/narwhals/extending/),
+meaning that libraries can become compatible without having to asking any other library
```
Comment (rgommers) [inline apps/labs/posts/dataframe-interop-pycon-lit-2024.md:142]:
```suggestion
meaning that libraries can become compatible without having to ask any other library
```

File: apps/labs/posts/dataframe-interop-pycon-lit-2024.md [hunk 8]
```diff
@@ -0,0 +1,180 @@
...
+A related project is Ibis, which [presents itself as a dataframe standard](https://voltrondata.com/resources/open-source-standards),
+and dispatches to 20+ backends. Some differences with Narwhals are:
+
+- Narwhals is ~1000 times lighter
+- Narwhals only supports 4 backends, Ibis more than 20
+- Narwhals is new, whereas Ibis is mature and production-ready
+
+The first point is a major differentiator:
+if your objective is to make your dataframe library dataframe-agnostic, and package size is a concern to you,
+then Narwhals is currently (as of April 2024) a more realistic solution - Ibis alone (310 MB) would put you above
```
Comment (rgommers) [inline apps/labs/posts/dataframe-interop-pycon-lit-2024.md:156]: "Ibis alone" should probably state that it's Ibis plus all its required dependencies?

File: apps/labs/posts/dataframe-interop-pycon-lit-2024.md [hunk 9]
```diff
@@ -0,0 +1,180 @@
...
+the AWS Lambda package size limit (250MB).
+
+![Image relative sizes of Narwhals (0.3 MB) vs Ibis (310 MB)](/posts/dataframe-interop-pycon-lit-2024/narwhals_vs_ibis.png)
```
Comment (rgommers) [inline apps/labs/posts/dataframe-interop-pycon-lit-2024.md:159]: This image is very rough, I suggest to remove it.

File: apps/labs/posts/dataframe-interop-pycon-lit-2024.md [hunk 10]
```diff
@@ -0,0 +1,180 @@
...
+
+Note that the projects are not in competition and have different goals.
```
Comment (rgommers) [inline apps/labs/posts/dataframe-interop-pycon-lit-2024.md:161]: This leaves the reader wondering quite a bit. Maybe add one sentence for each?

File: apps/labs/posts/dataframe-interop-pycon-lit-2024.md [hunk 11]
```diff
@@ -0,0 +1,180 @@
...
+of round-tripping back to your starting dataframe class.
+
+Nonetheless, does it work? Is it reliable?
+
+- Converting to pandas: reliable enough after pandas 2.0.2, though may sometimes
+  raise unnecessarily in some versions. It's already used by the `seaborn` and
+  `plotly` dataframe libraries.
+- Converting from pandas: unreliable, don't use it. There are cases when the results
+  are literal nonsense. These have generally been fixed, and will be available in
+  pandas 3.0+. For now, however, if you're using the interchange protocol to convert
```
Comment (rgommers) [inline apps/labs/posts/dataframe-interop-pycon-lit-2024.md:101]:
```suggestion
 pandas 3.0. For now, however, if you're using the interchange protocol to convert
```

**Review by rgommers — CHANGES_REQUESTED** [review]
Thanks Marco. This is a nice high-level story. A few suggestions, both content-wise and copy-editing.

**Comment by MarcoGorelli** [comment]
thanks Ralf for your review! I think I've addressed all comments, and I removed the Ibis comparison entirely

**Review by rgommers — APPROVED** [review]
Looks good to me now, thanks Marco! In it goes 🚀

**Pull request merged**